plugins/buildClusters/update.js

```diff
@@ -0,0 +1,125 @@
+        plugins: {
+            'hapi-swagger': {
+                security: [{ token: [] }]
+            }
+        },
+        handler: (request, reply) => {
+            const buildClusterFactory = request.server.app.buildClusterFactory;
+            const userFactory = request.server.app.userFactory;
+            const scm = buildClusterFactory.scm;
+            const name = request.params.name; // name of build cluster to update
+            const username = request.auth.credentials.username;
+            const scmContext = request.auth.credentials.scmContext;
+            const scmOrganizations = request.payload.scmOrganizations;
+
+            // Check permissions
+            // Must be Screwdriver admin to add Screwdriver build cluster
+            if (request.payload.managedByScrewdriver) {
+                const adminDetails = request.server.plugins.banners
+                    .screwdriverAdminDetails(username, scmContext);
+
+                if (!adminDetails.isAdmin) {
+                    return reply(boom.forbidden(
+                        `User ${adminDetails.userDisplayName}
+                        does not have Screwdriver administrative privileges.`
+                    ));
+                }
+
+                return buildClusterFactory.list({
+                    params: {
+                        name
+                    }
+                })
+                    .then((buildCluster) => {
+                        if (!buildCluster) {
+                            throw boom.notFound(`Build cluster ${name} does not exist`);
+                        }
+
+                        Object.assign(buildCluster, request.payload);
+
+                        return buildCluster.update()
+                            .then(updatedBuildCluster =>
+                                reply(updatedBuildCluster.toJson()).code(200)
+                            );
+                    })
+                    .catch(err => reply(boom.boomify(err)));
+            }
+
+            // Must have admin permission on org if adding org-specific build cluster
+            if (scmOrganizations && scmOrganizations.length === 0) {
+                return reply(boom.boomify(boom.badData(
+                    `No scmOrganizations provided for build cluster ${name}.`
+                )));
+            }
+
+            return userFactory.get({ username, scmContext })
+                .then(user => Promise.all([
+                    user.unsealToken(),
+                    buildClusterFactory.list({
+                        params: {
+                            name
+                        }
+                    })])
+                    .then(([token, buildCluster]) => {
+                        if (!buildCluster) {
+                            throw boom.notFound(`Build cluster ${name} does not exist`);
+                        }
+
+                        // To update scmOrganizations, user need to have admin permissions on both old and new organizations
```

Comment:
What if they are trying to remove an old org or it was renamed?

Reply:
Discussed with Min and we should just document that you need to first delete the old org in one update call to the Screwdriver API, then actually update the org name in Git, then add the new org in an update call to the Screwdriver API

Not sure if SD admins should have access to do everything or not.